src/sentry/search/snuba/backend.py

```diff
@@ -533,6 +534,28 @@ def _get_queryset_conditions(
             queryset_conditions.update(
                 {"issue.type": QCallbackCondition(lambda types: Q(type__in=types))}
             )
+            queryset_conditions.update(
+                {
+                    "message": QCallbackCondition(
+                        lambda query: Q(type=GroupType.ERROR.value)
+                        | (
+                            Q(
+                                type__in=(
+                                    GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value,
+                                    GroupType.PERFORMANCE_N_PLUS_ONE.value,
+                                    GroupType.PERFORMANCE_SLOW_SPAN.value,
+                                    GroupType.PERFORMANCE_SEQUENTIAL_SLOW_SPANS.value,
+                                    GroupType.PERFORMANCE_LONG_TASK_SPANS.value,
+                                    GroupType.PERFORMANCE_RENDER_BLOCKING_ASSET_SPAN.value,
+                                    GroupType.PERFORMANCE_DUPLICATE_SPANS.value,
+                                    GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value,
+                                )
+                            )
+                            & Q(message__icontains=query)
```

Comment:
Hopefully, this query (where message LIKE '%<query>%') on Groups table will be pared down enough to not scan a ton of data.